airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java

```diff
@@ -237,6 +239,7 @@ public void jobFailure(final JobFailureInput input) {
       job.getLastFailedAttempt().flatMap(Attempt::getFailureSummary)
           .ifPresent(failureSummary -> jobErrorReporter.reportSyncJobFailure(connectionId, failureSummary, jobContext));
     } catch (final IOException e) {
+      trackCompletionForInternalFailure(input.getJobId(), input.getConnectionId(), input.getAttemptNumber(), JobStatus.FAILED, e);
```

Comment:
should we move the `trackCompletion` call that appears in the try block to the last line? In case we see errors after `trackCompletion` succeeds -- we might want to avoid duplicate sync tracking (though maybe data team can dedupe?)

Reply:
@alovew Good call.  I will make sure that all calls to `trackCompletion` are last to avoid duplication.